### PR TITLE
BYTEA escape format support

### PR DESCRIPTION
**Pull request by BillyYccc**
fixes https://github.com/reactiverse/reactive-pg-client/issues/234.

### CODE REVIEW

File: src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java [hunk 1]
```diff
@@ -1287,6 +1293,45 @@ private static byte decodeHexChar(byte ch) {
     return (byte)(((ch & 0x1F) + ((ch >> 6) * 0x19) - 0x10) & 0x0F);
   }
 
+  private static boolean isRegardedAsHex(int index, int len, ByteBuf buff) {
```
Comment (vietj) [inline src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java]: rename `isHexFormat`

File: src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java [hunk 2]
```diff
@@ -1008,8 +1008,14 @@ private static OffsetDateTime textDecodeTIMESTAMPTZ(int index, int len, ByteBuf
   }
 
   private static Buffer textDecodeBYTEA(int index, int len, ByteBuf buff) {
-    // Shift 2 bytes: skip \x prolog
-    return Buffer.buffer(decodeHexStringToBytes(index + 2, len - 2, buff));
+    if (isRegardedAsHex(index, len, buff)) {
+      // hex format
+      // Shift 2 bytes: skip \x prolog
+      return Buffer.buffer(decodeHexStringToBytes(index + 2, len - 2, buff));
+    }else {
```
Comment (vietj) [inline src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java]: missing whitespace between } and else

File: src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java [hunk 3]
```diff
@@ -1287,6 +1293,45 @@ private static byte decodeHexChar(byte ch) {
...
+          int medium = buff.getByte(pos + index + 2) << 3;
+          int low = buff.getByte(pos + index + 3);
+          int escapedValue = high + medium + low;
+          if (escapedValue >= 32 && escapedValue <= 126) {
+            //printable escaped value
+            buffer.appendByte((byte) escapedValue);
+            pos += 4;
+            continue;
+          }
+        }
```
Comment (vietj) [inline src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java]: no else clause ?

**Comment by BillyYccc** [comment]
It looks like I have missed transforming non-printable octets to unescaped raw bytes, I'm now assuming executing `SELECT '\\251\251'::bytea;` should output bytes of `0x5C 0x32 0x35 0x31 0xA9`, need some changes to adapt to this.

**Comment by BillyYccc** [comment]
I have made a revision in https://github.com/reactiverse/reactive-pg-client/pull/241/commits/1e6f43a4cb32ffa868e10c829f1c111369d46352

File: src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java [hunk 4]
```diff
@@ -1287,6 +1293,48 @@ private static byte decodeHexChar(byte ch) {
     return (byte)(((ch & 0x1F) + ((ch >> 6) * 0x19) - 0x10) & 0x0F);
   }
 
+  private static boolean isHexFormat(int index, int len, ByteBuf buff) {
```
Comment (vietj) [inline src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java]: do we have test with empty bytea string

Comment (BillyYccc) [reply src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java]: added in https://github.com/reactiverse/reactive-pg-client/pull/241/commits/4b002d82a99295723a2427ea289e78a2558005ed

**Pull request closed without merging**